### PR TITLE
Add build message skeleton for incremental build.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @herohua @hellosnow @qinezh 
#449

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 1]
```diff
@@ -88,19 +92,21 @@ private Manifest BuildCore(DocumentBuildParameters parameters)
                         ManifestFile = "manifest",
                         XRefSpecMapFile = "xrefspecmap",
                         BuildModelManifestFile = "buildmodelmanifest",
-                        PostBuildModelManifestFile = "postbuildmodelmanifest",
+                        BuildMessageFile = "buildmessage",
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: >buildmessage [](start = 44, length = 12)

buildmessagefile?

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 2]
```diff
@@ -207,13 +213,13 @@ private bool GetCanIncremental(string configHash, string versionName)
             {
                 return false;
             }
-            var version = LastBuildInfo.Versions.SingleOrDefault(v => v.VersionName == versionName);
-            if (version == null)
+            _lbv = LastBuildInfo.Versions.Single(v => v.VersionName == versionName);
+            if (_lbv == null)
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: >_lbv [](start = 16, length = 4)

_lbv could not be null?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: should be SingleOrDefault

---
In reply to: [79977485](https://github.com/dotnet/docfx/pull/724#discussion_r79977485) [](ancestors = 79977485)

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: hard code message file name, will be remove.

---
In reply to: [79969979](https://github.com/dotnet/docfx/pull/724#discussion_r79969979) [](ancestors = 79969979)

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/BuildVersionInfo.cs [hunk 3]
```diff
@@ -80,32 +76,32 @@ public class BuildVersionInfo
         /// deserialized build model manifest
         /// </summary>
         [JsonIgnore]
-        public ModelManifest BuildModelManifest { get; set; }
+        public ModelManifest BuildModelManifest { get; } = new ModelManifest();
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/BuildVersionInfo.cs]: > public ModelManifest BuildModelManifest { get; } = new ModelManifest(); [](start = 7, length = 72)

remove

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 4]
```diff
@@ -79,28 +83,25 @@ private Manifest BuildCore(DocumentBuildParameters parameters)
...
                     var fileAttributes = ComputeFileAttributes(parameters);
-                    CurrentBuildInfo.Versions.Add(new BuildVersionInfo
+                    _cbv = new BuildVersionInfo
                     {
                         VersionName = parameters.VersionName,
                         ConfigHash = configHash,
                         AttributesFile = "attributes",
                         DependencyFile = "dependency",
                         ManifestFile = "manifest",
                         XRefSpecMapFile = "xrefspecmap",
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: how about we move these assignment to the definition like "build.info"?

**Comment by ansyral** [comment]
:shipit:

**Pull request merged**